File: backend/Mavex-Medafrica/mavex/src/main/java/com/medafrica/mavex/service/interfaces/NotificationTypeService.java
package com.medafrica.mavex.service.interfaces;

import com.medafrica.mavex.dto.email.NotificationTypeDTO;

import java.util.List;

public interface NotificationTypeService {

    List<NotificationTypeDTO> getAll();

    NotificationTypeDTO create(String name);

    NotificationTypeDTO update(Long id, String name);

    void delete(Long id);
}

File: frontend/mavex-frontend/src/app/core/services/notification-type.service.ts
import { Injectable, inject } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { NotificationType } from '../models/notification-type.model';

const BASE = '/api/notification-types';

@Injectable({ providedIn: 'root' })
export class NotificationTypeService {
  private readonly http = inject(HttpClient);

  getAll(): Observable<NotificationType[]> {
    return this.http.get<NotificationType[]>(BASE);
  }

  create(name: string): Observable<NotificationType> {
    return this.http.post<NotificationType>(BASE, { name });
  }

  update(id: number, name: string): Observable<NotificationType> {
    return this.http.put<NotificationType>(`${BASE}/${id}`, { name });
  }

  delete(id: number): Observable<void> {
    return this.http.delete<void>(`${BASE}/${id}`);
  }
}
